tests/test_requests.py

```diff
@@ -344,6 +345,38 @@ def test_param_cookiejar_works(self, httpbin):
         # Make sure the cookie was sent
         assert r.json()['cookies']['foo'] == 'bar'
 
+    def test_cookielib_cookiejar_on_redirect(self, httpbin):
+        """Tests resolve_redirect doesn't fail when merging cookies
+        with non-RequestsCookieJar cookiejar.
+
+        See GH #3579
+        """
+        cj = cookielib.CookieJar()
+        cookiejar_from_dict({'foo': 'bar'}, cj)
```

Comment:
While I know that `cookiejar_from_dict` mutates the cookiejar, not everyone will: can you change this to `cj = cookiejar_fro_dict({'foo': 'bar'}, cj)`? It's just slightly clearer.